lib/auth0/api/authentication_endpoints.rb

```diff
@@ -7,35 +7,48 @@ module AuthenticationEndpoints
       UP_AUTH = 'Username-Password-Authentication'.freeze
       JWT_BEARER = 'urn:ietf:params:oauth:grant-type:jwt-bearer'.freeze
 
-      # Retrives an access token
-      # @see https://auth0.com/docs/auth-api#!#post--oauth-access_token
+      # Given the social provider's access_token and the connection, this endpoint will authenticate the user
+      # with the provider and return a JSON with the access_token and, optionally, an id_token
+      # @see https://auth0.com/docs/api/authentication#social-with-provider-s-access-token
       # @param access_token [string] Social provider's access_token
       # @param connection [string] Currently, this endpoint only works for Facebook, Google, Twitter and Weibo
+      # @param scope [string] Use openid to get an id_token, or openid profile email to include user information
+      #   in the id_token. If null, only an access_token will be returned.
+      # @return [json] Returns the access token
+      def obtain_access_token(access_token, connection = 'facebook', scope = 'openid')
+        raise Auth0::InvalidParameter, 'Must supply a valid code' if access_token.to_s.empty?
+        request_params = { client_id: @client_id, access_token: access_token, connection: connection, scope: scope }
+        post('/oauth/access_token', request_params)['access_token']
+      end
+
+      # This is the OAuth 2.0 grant that server processes utilize in order to access an API.
+      # Use this endpoint to directly request an access_token by using the Client Credentials
+      # @see https://auth0.com/docs/api/authentication#client-credentials
+      # @param audience [string] The unique identifier of the target API you want to access.
       # @return [json] Returns the access token
-      def obtain_access_token(access_token = nil, connection = 'facebook', scope = 'openid')
-        if access_token
-          request_params = { client_id: @client_id, access_token: access_token, connection: connection, scope: scope }
-          post('/oauth/access_token', request_params)['access_token']
-        else
-          request_params = { client_id: @client_id, client_secret: @client_secret, grant_type: 'client_credentials' }
-          post('/oauth/token', request_params)['access_token']
-        end
+      def client_credentials(audience = nil)
+        request_params = {
+          client_id:     @client_id,
+          client_secret: @client_secret,
+          grant_type:    'client_credentials',
+          audience:      audience
+        }
+        post('/oauth/token', request_params)['access_token']
```

Comment:
Let's return the complete response since it could have other tokens or even useful attributes like `expires_in`